File: NutriSmart/app/src/main/java/com/example/nutrismart/data/repository/AuthRepository.kt
package com.example.nutrismart.data.repository

import com.example.nutrismart.data.SessionManager
import com.example.nutrismart.data.model.*
import com.example.nutrismart.data.remote.NutriSmartApi
import retrofit2.Response
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AuthRepository @Inject constructor(
    private val api: NutriSmartApi,
    private val sessionManager: SessionManager
) {
    suspend fun login(request: AuthRequest): Response<AuthResponse> {
        return api.login(request)
    }

    suspend fun googleLogin(request: GoogleLoginRequest): Response<AuthResponse> {
        return api.googleLogin(request)
    }

    suspend fun register(request: RegisterRequest): Response<Void> {
        return api.register(request)
    }

    suspend fun verify(request: VerifyRequest): Response<AuthResponse> {
        return api.verify(request)
    }

    suspend fun resendCode(email: String): Response<Void> {
        return api.resendCode(email)
    }

    fun saveAuthData(authData: AuthResponse) {
        sessionManager.saveAuthToken(authData.token)
        sessionManager.saveUserId(authData.userId)
        sessionManager.saveProfileComplete(authData.isProfileComplete)
        sessionManager.saveIsVerified(authData.isVerified)
    }

    fun saveIsGoogleUser(isGoogle: Boolean) {
        sessionManager.saveIsGoogleUser(isGoogle)
    }

    fun getAuthToken(): String? = sessionManager.fetchAuthToken()
    fun getUserId(): Long = sessionManager.fetchUserId()
    fun isProfileComplete(): Boolean = sessionManager.isProfileComplete()
    fun isVerified(): Boolean = sessionManager.isVerified()
    fun getWakeUpTime(): String = sessionManager.getWakeUpTime()
    
    fun saveWakeUpTime(time: String) {
        sessionManager.saveWakeUpTime(time)
    }
    
    fun clearSession() {
        sessionManager.clearSession()
    }
}

File: NutriSmartIOS/data/repository/AuthRepository.swift
//
//  AuthRepository.swift
//  NutriSmartIOS
//

import Foundation

class AuthRepository {
    static let shared = AuthRepository()
    
    private let api = NutriSmartApiService.shared
    private let sessionManager = SessionManager.shared
    
    private init() {}
    
    func login(request: AuthRequest) async throws -> AuthResponse {
        return try await api.login(request: request)
    }
    
    func googleLogin(request: GoogleLoginRequest) async throws -> AuthResponse {
        return try await api.googleLogin(request: request)
    }
    
    func register(request: RegisterRequest) async throws -> EmptyResponse {
        return try await api.register(request: request)
    }
    
    func verify(request: VerifyRequest) async throws -> AuthResponse {
        return try await api.verify(request: request)
    }
    
    func resendCode(email: String) async throws -> EmptyResponse {
        return try await api.resendCode(email: email)
    }
    
    func saveAuthData(authData: AuthResponse) {
        sessionManager.saveAuthToken(authData.token)
        sessionManager.saveUserId(authData.userId)
        sessionManager.saveProfileComplete(authData.isProfileComplete)
        sessionManager.saveIsVerified(authData.isVerified)
        
        UserSession.shared.token = authData.token
        UserSession.shared.currentUserId = authData.userId
    }
    
    func saveIsGoogleUser(_ isGoogle: Bool) {
        sessionManager.saveIsGoogleUser(isGoogle)
        UserSession.shared.isGoogleUser = isGoogle
    }
    
    func getAuthToken() -> String? {
        return sessionManager.fetchAuthToken()
    }
    
    func getUserId() -> Int {
        return sessionManager.fetchUserId()
    }
    
    func isProfileComplete() -> Bool {
        return sessionManager.isProfileComplete()
    }
    
    func isVerified() -> Bool {
        return sessionManager.isVerified()
    }
    
    func getWakeUpTime() -> String {
        return sessionManager.getWakeUpTime()
    }
    
    func saveWakeUpTime(_ time: String) {
        sessionManager.saveWakeUpTime(time)
    }
    
    func clearSession() {
        sessionManager.clearSession()
        UserSession.shared.clear()
    }
}
